internal/search/query/types.go

```diff
@@ -59,26 +65,172 @@ func (q OrdinaryQuery) IsCaseSensitive() bool {
 	return q.Query.IsCaseSensitive()
 }
 
-// AndOrQuery satisfies the interface for QueryInfo with empty values. Its
-// methods are not currently used.
-func (AndOrQuery) RegexpPatterns(field string) (values, negatedValues []string) {
-	return nil, nil
+// AndOrQuery satisfies the interface for QueryInfo with unvalidated string
+// values. These methods and dependent functions are only callable via an
+// andOrQuery site flag and not intended for use.
+func (q AndOrQuery) RegexpPatterns(field string) (values, negatedValues []string) {
+	VisitField(q.query, field, func(visitedValue string, negated bool) {
+		if negated {
+			negatedValues = append(negatedValues, visitedValue)
+		} else {
+			values = append(values, visitedValue)
+		}
+	})
+	log15.Info("Query", "RegexpPatterns", field)
```

Comment:
```suggestion
	log15.Debug("Query", "RegexpPatterns", field)
```

But maybe fine for now if all new methods in the PR are only being called behind a feature flag.

Reply:
Yeah, going to just keep this as info for now.